Dockerfile

```diff
@@ -43,13 +43,10 @@ RUN case ${TARGETPLATFORM} in \
     printf "Building OAuth2 Proxy for arch ${GOARCH}\n" && \
     GOARCH=${GOARCH} VERSION=${VERSION} make build && touch jwt_signing_key.pem
 
-# Copy binary to alpine
+# Copy binary to distroless
 FROM ${RUNTIME_IMAGE}
 COPY nsswitch.conf /etc/nsswitch.conf
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/oauth2-proxy /bin/oauth2-proxy
 COPY --from=builder /go/src/github.com/oauth2-proxy/oauth2-proxy/jwt_signing_key.pem /etc/ssl/private/jwt_signing_key.pem
 
-# UID/GID 65532 is also known as nonroot user in distroless image
-USER 65532:65532
```

Comment:
Since we're also going to provide an alpine image, what would be the best solution here?

Reply:
Fine with me. If people want to run it as nonroot with the "new" alpine images. They should explicitly set the user in the kubernetes manifest. We should maybe add a comment about that in the documentation and inside the helm chart.

Reply:
I'll do a quick test in a few minutes.

Reply:
So removing the user id doesn't break anything. As it doesn't exist in alpine anyway.
![image](https://github.com/oauth2-proxy/oauth2-proxy/assets/18513179/ed0d59ce-eafc-4bb7-97cc-4418e828415e)

Reply:
user 65534 would be a valid user id in alpine without creating a custom user like the distroless project does for 65532.

Reply:
Therefore, if people still want to use the alpine based image in the future and have restrictions caused by kubernetes security policies they just have to specify the proper securityContext. Something like:

```yaml
securityContext:
  runAsUser: 65532
  runAsGroup: 65532
```